clang/lib/Analysis/CallGraph.cpp

```diff
@@ -165,13 +164,14 @@ class CGBuilder : public StmtVisitor<CGBuilder> {
     if (G->shouldSkipConstantExpressions()) {
       Decl **DeclPointer = DS->decls().begin();
       for (Stmt *SubStmt : DS->children()) {
-        const Decl *D = *DeclPointer;
         if (!isConstExprVar(*DeclPointer) && SubStmt)
```

Comment:
Sorry, one more, should have caught it the first time.
```suggestion
        if (!isConstExprVar(*DeclPointer))
```

Reply:
I kept this check to mirror what it done by default in `VisitDeclStmt`. Clang default `VisitDeclStmt `calls `VisitChildren `on L177 below. There is a check for `if (SubStmt)` here.

Reply:
Ah, because it could be nullptr but not the end of the range.   Got it, thanks.